pandas/core/generic.py

```diff
@@ -5891,6 +5891,7 @@ def astype(
 
         # GH 19920: retain column metadata after concat
         result = concat(results, axis=1, copy=False)
+        result = self._constructor(result)  # GH#40810 retain subclass
```

Comment:
copy=False?

Reply:
doesn't make a difference, goes through the fastpath either way